c/number.c

```diff
@@ -637,7 +637,7 @@ static ptr big_mul(tc, x, y, xl, yl, sign) ptr tc, x, y; iptr xl, yl; IBOOL sign
   return copy_normalize(&BIGIT(W(tc),0),xl+yl,sign);
 }
 
-#define SHORTMIN (most_negative_fixnum / (1 << (fixnum_bits / 2)))
+#define SHORTMIN ((most_negative_fixnum / (1 << (fixnum_bits / 2))) + 1)
```

Comment:
Is it worth doing something to make the range tighter for 32-bit systems? Maybe something like
<pre>#define SHORTMIN ((most_negative_fixnum / (1 << (fixnum_bits / 2))) + (fixnum_bits % 1))</pre>

Reply:
I suppose on 32-bit systems SHORTMIN = -23170, which is approx. -sqrt(2^29-1)

Reply:
And SHORTMAX = 23170 on 32-bit systems.

Reply:
<pre>/* SHORTRANGE is -floor(sqrt(most_positive_fixnum))..floor(sqrt(most_positive_fixnum)).
   We don't use sqrt because it rounds up for fixnum_bits = 61 */
#if (fixnum_bits == 30)
#define SHORTRANGE(x) (-23170 <= (x) && (x) <= 23170)
#elif (fixnum_bits == 61)
#define SHORTRANGE(x) (-0x3FFFFFFF <= (x) && (x) <= 0x3FFFFFFF)
#endif
</pre>